### PR TITLE
Recipe- BeansXmlNamespace

### PR DESCRIPTION
**Pull request by ranuradh**
## What's changed?
image

image

### Checklist
- [X] I've added unit tests to cover both positive and negative cases
- [X] I've added the license header to any new files through `./gradlew licenseFormat`
- [X] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

File: src/main/java/org/openrewrite/java/migrate/BeansXmlNamespace.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                        nameSpaceValue = attribute.getValueAsString();
+                    } else if (attribute.getKeyAsString().endsWith("schemaLocation")) {
+                        schemaLocation = attribute.getValueAsString();
+                    } else if (DISABLE_VALIDATING.equalsIgnoreCase(attribute.getKeyAsString())) {
+                        return t;
+                    }
+
+                }
+                if (!checkNameSpaceSchemaLocation(nameSpaceValue, schemaLocation)) {
+                    return addAttribute(t, DISABLE_VALIDATING, String.valueOf(true), ctx);
```
Comment (timtebeek) [inline src/main/java/org/openrewrite/java/migrate/BeansXmlNamespace.java:69]: The way I read the screenshots attached to this PR `org.jboss.weld.xml.disableValidating` is a JVM property that you pass in at startup, not something to add to `beans.xml`. I could be mistaken as I lack background here, but figured ask before we continue with this PR.

Comment (ranuradh) [reply src/main/java/org/openrewrite/java/migrate/BeansXmlNamespace.java:69]: you are right @timtebeek  ..i think i understood this wrong..our fix is - ```The automated fix for this rule updates incompatible namespaces with the value appropriate to the specified schemaLocation```.  Will update the PR in a bit

**Comment by timtebeek** [comment]
So from what I understand now, we update the `:schemaLocation` to match what we find in `xmlns` inside any `beans.xml`?

File: src/main/java/org/openrewrite/java/migrate/BeansXmlNamespace.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    private static final XPathMatcher BEANS_MATCHER = new XPathMatcher("/beans");
+    private static final String NS_SUN = "http://java.sun.com/xml/ns/javaee";
+    private static final String NS_JCP = "http://xmlns.jcp.org/xml/ns/javaee";
+    private static final String SUN_SCHEMA_LOCATION = "http://java.sun.com/xml/ns/javaee http://java.sun.com/xml/ns/javaee/beans_1_0.xsd";
+    private static final String JCP_SCHEMA_LOCATION = "http://xmlns.jcp.org/xml/ns/javaee http://xmlns.jcp.org/xml/ns/javaee/beans_1_1.xsd";
+    private static final String DISABLE_VALIDATING = "org.jboss.weld.xml.disableValidating";
+
+    @Override
+    public String getDisplayName() {
+        return "Check valid namespace and schema location in the`beans.xml` file ";
```
Comment (timtebeek) [inline src/main/java/org/openrewrite/java/migrate/BeansXmlNamespace.java:39]: I'm not sure the display name and description fully match what the recipe now does, so I'll propose two changes for you to consider, adapt & adopt or reject.
```suggestion
        return "Change `beans.xml` `schemaLocation` to match XML namespace";
```

**Review by timtebeek — APPROVED** [review]
Let me know what you'd want to do with the display name and description; otherwise approved.

**Comment by ranuradh** [comment]
@timtebeek I have updated it

**Comment by timtebeek** [comment]
@ranuradh This recipe is not currently part of any on the larger declarative recipe migrations;
https://github.com/openrewrite/rewrite-migrate-java/tree/main/src/main/resources/META-INF/rewrite
Should this recipe be run in any of the migrations to Jakarta for instance?

**Comment by timtebeek** [comment]
I'm asking since on https://github.com/openrewrite/rewrite-migrate-java/pull/372#discussion_r1422533602 we identified that we can likely do the same with declarative recipes instead, which would be easier to maintain going forward.

**Pull request merged**